flake.nix

```diff
@@ -9,24 +9,31 @@
         nixpkgs = self;
       };
 
-      lib = import ./lib;
+      lib = (import ./lib).extend libVersionInfoOverlay;
+
+      libVersionInfoOverlay = finalLib: prevLib: {
+        trivial = prevLib.trivial // {
+          versionSuffix =
+            ".${finalLib.substring 0 8 (self.lastModifiedDate or self.lastModified or "19700101")}.${self.shortRev or "dirty"}";
+          version = finalLib.trivial.release + finalLib.trivial.versionSuffix;
+          revisionWithDefault = default: self.rev or default;
+        };
+      };
 
       forAllSystems = lib.genAttrs lib.systems.flakeExposed;
     in
     {
+      overlays.default = final: prev: {
+        lib = prev.lib.extend libVersionInfoOverlay;
+      };
```

Comment:
Fixed the naming of the overlay.

Won't resolve though until we're aligned on how to deal with `import inputs.nixpkgs`.

Reply:
Is the naming OK and the fact that you'd have to apply this manually when doing `import inputs.nixpkgs` @roberth? Also, anything else to discuss / fix? :)

Reply:
I think we should discourage `import inputs.nixpkgs`, because it just can't work well. It's just not something that can ever work as well as the actual flake outputs, unless flakes allows `outPath` to be a `__functor` function. I really don't know what to think of that... It should probably be an explicit feature of flakes and not something that allows arbitrary stuff to be put in `outPath`. And even then it's kind of bad because of `import "${inputs.nixpkgs}"`.

Ignoring all that ugliness for now, I think the `nixpkgs` should provide a proper `lib`-based entrypoint into Nixpkgs, just like we have `lib.nixosSystem`. That way it can actually work well, and not be a strange experience. (`import` a flake, really?)

Some possible names
- `lib.nixpkgs`: not a fan because Nixpkgs *should* be more than just the function that returns "pkgs"
- `lib.callNixpkgs`: arbitrary enough that we don't paint ourselves into a corner just yet

We should probably get the arguments right this time, along the lines of
`{ hostPlatform, buildPlatform ? hostPlatform, config , overlays }:` instead of the unholy mess that we have in `default.nix`. Maybe `default.nix` should stay bad, as a motivator for people to use the better entrypoint?

(As mentioned in the review, I think we should solve all these problems in a new PR after merging this one, and not have `overlays` in the outputs for this)